expression/builtin_miscellaneous.go

```diff
@@ -289,7 +289,52 @@ type builtinIsIPv4Sig struct {
 
 // See https://dev.mysql.com/doc/refman/5.7/en/miscellaneous-functions.html#function_is-ipv4
 func (b *builtinIsIPv4Sig) eval(row []types.Datum) (d types.Datum, err error) {
-	return d, errFunctionNotExists.GenByArgs("IS_IPV4")
+	args, err := b.evalArgs(row)
+	if err != nil {
+		return types.Datum{}, errors.Trace(err)
+	}
+	if args[0].IsNull() {
+		d.SetInt64(0)
+		return d, nil
+	}
+	// isIPv4(str)
+	// args[0] string
+	s, err := args[0].ToString()
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+	if mysqlIsIPv4(s) {
+		d.SetInt64(1)
+	} else {
+		d.SetInt64(0)
+	}
+	return d, nil
+}
+
+// mysql is_ipv4 don't accesp IPv6 expressed IPv4 format
+func mysqlIsIPv4(ip string) bool {
```

Comment:
1. I think it's not necessary to name it as mysqlXXX, `isIPv4` is enough~
2. You mean '::ffff:1.2.3.4' would be regarded as IPv4 address in Golang, but IPv6 in MySQL?
I think we can comment this function like 
`check whether IP is IPv4 address which format is 'A.B.C.D'(0<=A/B/C/D<=255)`